siva/metadata.go

```diff
@@ -0,0 +1,156 @@
+package siva
+
+import (
+	"io/ioutil"
+	"os"
+
+	"github.com/ghodss/yaml"
+	billy "gopkg.in/src-d/go-billy.v4"
+)
+
+const (
+	// LibraryMetadataFile is the name of the file that holds library metadata.
+	LibraryMetadataFile = "library.yaml"
+)
+
+// LibraryMetadata holds information about the library.
+type LibraryMetadata struct {
+	Version int `json:"version"`
+}
+
+// ParseLibraryMetadata parses the yaml representation of library metadata.
+func ParseLibraryMetadata(d []byte) (*LibraryMetadata, error) {
+	var m LibraryMetadata
+
+	err := yaml.Unmarshal(d, &m)
+	if err != nil {
+		return nil, err
+	}
+
+	return &m, nil
+}
+
+// LoadLibraryMetadata reads and parses a library metadata file.
+func LoadLibraryMetadata(fs billy.Filesystem) (*LibraryMetadata, error) {
```

Comment:
we could use call it `NewLibraryMetadata` to clarify it's a construtor. Also it could accept a path parameter so the path to Open is not harcoded to the constant `LibraryMetadataPath`

Reply:
Now `loadLibraryMetadata` is not exported. Metadata read and write is done internally. There's `NewLibraryMetadata` in case it is needed to bootstrap the structure externally.